rbac/management/workspace/service.py

```diff
@@ -102,6 +105,50 @@ def destroy(self, instance: Workspace) -> None:
         dual_write_handler.replicate_deleted_workspace()
         instance.delete()
 
+    def move(self, instance: Workspace, request: Request) -> Response:
+        """Move a workspace under new parent."""
+        # Parent id validations
+        new_parent_id = request.data.get("parent_id")
+        if not new_parent_id:
+            raise serializers.ValidationError({"parent_id": "The 'parent_id' field is required."})
+        validate_uuid(new_parent_id)
+
+        # Parent workspace validations
+        tenant = request.tenant
+        new_parent_workspace = Workspace.objects.filter(id=new_parent_id, tenant=tenant).first()
+        if not new_parent_workspace:
+            raise serializers.ValidationError(
+                {"parent_id": f"Parent workspace '{new_parent_id}' doesn't exist in tenant"}
+            )
+
+        # Prevent moving workspace under itself
```

Comment:
Maybe just preference, but for readability/testability, what do you think about breaking these conditions out into self-documenting methods similar to `_enforce_hierarchy_depth` and others? If the names are intention-revealing, it would encapsulate the logic and remove the need for comments here as well.